packages/twenty-front/src/modules/ui/navigation/navigation-drawer/hooks/useWorkspaceSwitching.ts

```diff
@@ -0,0 +1,51 @@
+import { useNavigate } from 'react-router-dom';
+import { useRecoilState, useSetRecoilState } from 'recoil';
+
+import { currentWorkspaceMemberState } from '@/auth/states/currentWorkspaceMemberState';
+import { currentWorkspaceState } from '@/auth/states/currentWorkspaceState';
+import { tokenPairState } from '@/auth/states/tokenPairState';
+import { ColorScheme } from '@/workspace-member/types/WorkspaceMember';
+import { useGenerateJwtMutation } from '~/generated/graphql';
+
+export const useWorkspaceSwitching = () => {
+  const navigate = useNavigate();
+  const setTokenPair = useSetRecoilState(tokenPairState);
+  const [generateJWT] = useGenerateJwtMutation();
+  const [currentWorkspace, setCurrentWorkspace] = useRecoilState(
+    currentWorkspaceState,
+  );
+  const setCurrentWorkspaceMember = useSetRecoilState(
+    currentWorkspaceMemberState,
+  );
+
+  const swtitchWorkspace = async (workspaceId: string) => {
```

Comment:
Small typo!

Reply:
You're right! Now we can pass either an IconComponent or avatar :)